### ISSUE TITLE
Prompt flickers with every key input on Windows Terminal

### ISSUE THREAD

**Issue by abhbh** [nushell/nushell#13440]
### Describe the bug

With every keystroke, the default nu prompt flickers. Issue only happens on Windows Terminal. Wezterm, alacritty is fine. All other shells (Powershell, cmd, bash) work fine in Windows Terminal too.

### How to reproduce

1. Create a new Windows Terminal Profile for nu
2. Delete nushell config folder to use stock settings
3. With no custom config, the prompt line flickers with every keystroke

### Expected behavior

Behave normally without flickering

### Configuration

Stock config with just released 0.96.0

### Additional context

Last good version is 0.93.0. Windows Terminal is the latest v1.20

The issue begun to happen in 0.94.0.

**Comment by DerTee** [comment]
Can't reproduce with the following config (fairly similar to OP except my Graphics card is much older):
Nushell 0.96.0 x86_64-pc-windows-msvc
Windows Terminal Version: 1.20.11781.0
Windows 11 Home Version 23H2 OS build 22631.3880
GeForce GTX 1060 NVidia Driver Version: 560.70

Have you tried toggling `Terminal > Settings > Rendering > Use new text renderer ("AtlasEngine")` ? I did try it out in both on and off and it made no difference, but maybe it does for you.

**Comment by abhbh** [comment]
To no avail with that toggle.

I also tried it with all default settings in Windows Terminal. I'll also try it in a VM to check if it's hardware dependent or to my setup.

Also, it's only the prompt that flickers (specifically the `chevron>` and the time on right). The previous prompts and the scrollback text does not.

**Comment by DerTee** [comment]
I'll ask like an idiot: Does the time or anything else update for you in the current prompt? It doesn't for me, it's static.

**Comment by fdncred** [comment]
May be related
https://github.com/microsoft/terminal/issues/16769
https://github.com/microsoft/terminal/issues/13710

I'd also play with the shell_integration functionality in the config.nu to see if changing any of those settings help.

**Comment by abhbh** [comment]
> I'll ask like an idiot: Does the time or anything else update for you in the current prompt? It doesn't for me, it's static.

No. It is supposed to just show the timestamp of the last command run.

> I'll also try it in a VM to check if it's hardware dependent or to my setup.

I cannot reproduce the issue in a new Windows VM although WT is running here without Hardware Acceleration. I will check again in another Windows 11 23H2 machine if I can find one.

**Comment by abhbh** [comment]
> I will check again in another Windows 11 23H2 machine if I can find one.

I can reproduce this issues on another laptop but this time with Intel Graphics. The flickering here is a bit less frequent. Every 4-5 letters but still happens.

Stock config. Nu 0.96.1 on Windows 11 23H2 and WT 1.20 with all updates applied.

**Comment by fdncred** [comment]
Have you tried this suggestions?

> I'd also play with the shell_integration functionality in the config.nu to see if changing any of those settings help.

**Comment by MadSpindel** [comment]
I changed osc133 to false and now it doesn't flicker anymore for me. According to the config, this setting `is used to enable terminals to know where the prompt is, the command is, where the command finishes, and where the output of the command is`

**Comment by abhbh** [comment]
> I changed osc133 to false and now it doesn't flicker anymore for me.

Yes. That fixed it.

In versions not affected, shell integration was false by default. Other terminals work fine with it set to true. I tested Alacritty and Wezterm.

The https://github.com/microsoft/terminal/issues/16769 issue was closed after a fix but the fix was reverted in a later build.
